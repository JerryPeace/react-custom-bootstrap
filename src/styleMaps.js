const styleMaps = {
    CLASSES: {
        'alert': 'alert',
        'badge': 'badge',
        'banner': 'banner',
        'brand': 'brand',
        'breadcrumb': 'breadcrumb',
        'button': 'btn',
        'btn-icon': 'btn-icon',
        'button-group': 'btn-group',
        'btn-content': 'btn-content',
        'button-toolbar': 'btn-toolbar',
        'caret': 'caret',
        'checkbox': 'checkbox',
        'checkbox-radio-label': 'checkbox-radio-label',
        'check-style': 'check-style',
        'collapse': 'collapse',
        'collapse-group': 'collapse-group',
        'column': 'col',
        'control-group': 'form-group',
        'control-label': 'control-label',
        'controls': 'controls',
        'divider': 'divider',
        'dropdown': 'dropdown',
        'dropdown-toggle': 'dropdown-toggle',
        'dropdown-menu': 'dropdown-menu',
        'dropdown-menu-right': 'dropdown-menu-right',
        'dropdown-header': 'dropdown-header',
        'form': 'form',
        'form-actions': 'form-actions',
        'form-horizontal': 'form-horizontal',
        'glyphicon': 'glyphicon',
        'checkbox-group': 'checkbox-group',
        'radio-group': 'radio-group',
        'help-block': 'help-block',
        'has-success': 'has-success',
        'has-warning': 'has-warning',
        'has-error': 'has-error',
        'inline': 'inline',
        'input': 'input',
        'input-group': 'input-group',
        'label': 'label',
        'label-field': 'label-field',
        'list-group-item': 'list-group-item',
        'loader-content': 'loader-content',
        'logo': 'logo',
        'modal': 'modal',
        'modal-body': 'modal-body',
        'modal-content': 'modal-content',
        'modal-dialog': 'modal-dialog',
        'modal-footer': 'modal-footer',
        'modal-header': 'modal-header',
        'modal-title': 'modal-title',
        'modal-icon-info': 'modal-icon-info',
        'modal-icon-warn': 'modal-icon-warning',
        'modal-icon-success': 'modal-icon-success',
        'modal-icon-error': 'modal-icon-error',
        'multiple-selection': 'multiple-selection',
        'nav': 'nav',
        'navbar': 'navbar',
        'navbar-nav': 'navbar-nav',
        'pagination': 'pagination',
        'panel': 'panel',
        'panel-body': 'panel-body',
        'panel-collapse': 'panel-collapse',
        'panel-default': 'panel-default',
        'panel-group': 'panel-group',
        'panel-heading': 'panel-heading',
        'panel-title': 'panel-title',
        'progress-bar': 'progress-bar',
        'radio': 'radio',
        'row': 'row',
        'select': 'select',
        'tab-content': 'tab-content',
        'tab-pane': 'tab-pane',
        'tabbable': 'tabbable',
        'thumbnail': 'thumbnail',
        'tm-loader': 'tm-loader',
        'tree-dropdown': 'tree-dropdown',
        'well': 'well',
        'tm': 'tm',
        'collapser': 'collapser',
        'collapser-title': 'collapser-title',
        'arrow': 'arrow',
        'arrow-icon': 'fa fa-caret-right'
    },
    STYLES: [
        'ban-circle',
        'block',
        'circle-arrow-down',
        'danger',
        'default',
        'download-alt',
        'error',
        'header',
        'important',
        'info',
        'inline',
        'inverse',
        'link',
        'list',
        'ok',
        'ok-circle',
        'pills',
        'primary',
        'search',
        'star',
        'success',
        'tabs',
        'trash',
        'user',
        'warning',
        'horizontal',
        'exclamation-sign',
        'plain',
        'blue',
        'green',
        'cyan',
        'orange',
        'red',
        'indigo',
        'yellow',
        'light-gray',
        'gray',
        'dark'
    ],
    addStyle(name) {
        styleMaps.STYLES.push(name);
    },
    SIZES: {
        'large': 'large',
        'lg': 'lg',
        'md': 'md',
        'medium': 'medium',
        'sm': 'sm',
        'small': 'small',
        'xlarge': 'xlarge',
        'xs': 'xs',
        'xsmall': 'xs',
    },
    GRID_COLUMNS: 12
};

export default styleMaps;
