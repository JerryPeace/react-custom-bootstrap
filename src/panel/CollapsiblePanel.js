import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import Collapse from '../collapse/Collapse';
import styleMaps from '../styleMaps';

class CollapsiblePanel extends Collapse {

    static propTypes = {
        titleText: PropTypes.node
    };

    constructor(props) {
        super(props);
        this.handleClick = ::this.handleClick;
    }

    renderTitle() {
        const { titleText } = this.props;
        const titleStyle = {
            fontSize: '16px',
            color: '#222'
        };
        return (
            <span
                className={`${styleMaps.CLASSES.collapse}-header-text`}
                style={titleStyle}>
                {titleText}
            </span>
        );
    }

    renderArrowIcon() {
        const arrowStyle = {
            float: 'right',
            fontSize: '20px',
            color: '#666'
        };
        const cls = classNames(`${styleMaps.CLASSES.collapse}-header-arrow-icon`, 'fa', {
            'fa-angle-down': this.state.in,
            'fa-angle-up': !this.state.in
        });

        return (<span className={cls} style={arrowStyle}></span>);
    }

    renderHeader() {
        const collapserCls = `${styleMaps.CLASSES.tm}-${styleMaps.CLASSES.collapser}`;
        const cls = classNames(`${styleMaps.CLASSES.collapse}-header`, {
            closed: !this.state.in
        }, collapserCls);
        const headerStyle = {
            padding: '8px 12px',
            backgroundColor: '#eee',
            height: '40px'
        };

        return (
            <div
                onClick={this.handleClick}
                className={cls}
                style={headerStyle}>
                {this.renderArrowIcon()}
                {this.renderTitle()}
            </div>
        );
    };

    renderPanelBody() {
        const { children } = this.props;
        const clsName = styleMaps.CLASSES.collapse;
        const cls = classNames({
            in: this.state.in
        }, clsName);
        const bodyStyle = { padding: '8px 0px' };

        return (
            <div className={cls}>
                <div style={bodyStyle}>
                    {children}
                </div>
            </div>
        );
    }
}

export default CollapsiblePanel;
