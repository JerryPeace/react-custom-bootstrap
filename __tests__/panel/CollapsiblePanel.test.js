import React from 'react';
import { mount } from 'enzyme';
import _ from 'lodash';
import CollapsiblePanel from 'panel/CollapsiblePanel';
import styleMaps from 'styleMaps';

describe('CollapsiblePanel', () => {
    test('render CollapsiblePanel default', () => {
        const collapsible = mount(
            <CollapsiblePanel titleText='test-title'>
                <div>test</div>
            </CollapsiblePanel>
        );
        expect(collapsible).toMatchSnapshot();
    });

    test('click on header', () => {
        const collapsible = mount(
            <CollapsiblePanel titleText='test-title'>
                <div>test</div>
            </CollapsiblePanel>
        );

        collapsible.find(`.${styleMaps.CLASSES.collapse}-header`).at(0).simulate('click');
        collapsible.update();
        expect(collapsible).toMatchSnapshot();
    });

    test('click on header text', () => {
        const collapsible = mount(
            <CollapsiblePanel titleText='test-title'>
                <div>test</div>
            </CollapsiblePanel>
        );

        collapsible.find(`.${styleMaps.CLASSES.collapse}-header-text`).at(0).simulate('click');
        collapsible.update();
        expect(collapsible).toMatchSnapshot();
    });

    test('click on header arrow', () => {
        const collapsible = mount(
            <CollapsiblePanel titleText='test-title'>
                <div>test</div>
            </CollapsiblePanel>
        );

        collapsible.find(`.${styleMaps.CLASSES.collapse}-header-arrow-icon`).at(0).simulate('click');
        collapsible.update();
        expect(collapsible).toMatchSnapshot();
    });

    test('click on header close by clicking on arrow and open on text', () => {
        const collapsible = mount(
            <CollapsiblePanel titleText='test-title'>
                <div>test</div>
            </CollapsiblePanel>
        );
        // close panel by clicking on arrow
        collapsible.find(`.${styleMaps.CLASSES.collapse}-header-arrow-icon`).at(0).simulate('click');
        collapsible.update();
        expect(collapsible.state('in')).toBeFalsy();
        expect(collapsible).toMatchSnapshot();
        // open panel by clicking on text
        collapsible.find(`.${styleMaps.CLASSES.collapse}-header-text`).at(0).simulate('click');
        collapsible.update();
        expect(collapsible.state('in')).toBeTruthy();
        expect(collapsible).toMatchSnapshot();
    });
});
