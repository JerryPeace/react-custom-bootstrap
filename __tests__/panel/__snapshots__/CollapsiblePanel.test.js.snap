// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsiblePanel click on header 1`] = `
<CollapsiblePanel
  collapsible={true}
  defaultExpand={true}
  titleText="test-title"
>
  <div
    className="panel-collapse"
  >
    <div>
      <div
        className="collapse-header closed tm-collapser"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#eee",
            "height": "40px",
            "padding": "8px 12px",
          }
        }
      >
        <span
          className="collapse-header-arrow-icon fa fa-angle-up"
          style={
            Object {
              "color": "#666",
              "float": "right",
              "fontSize": "20px",
            }
          }
        />
        <span
          className="collapse-header-text"
          style={
            Object {
              "color": "#222",
              "fontSize": "16px",
            }
          }
        >
          test-title
        </span>
      </div>
      <div
        className="collapse"
      >
        <div
          style={
            Object {
              "padding": "8px 0px",
            }
          }
        >
          <div>
            test
          </div>
        </div>
      </div>
    </div>
  </div>
</CollapsiblePanel>
`;

exports[`CollapsiblePanel click on header arrow 1`] = `
<CollapsiblePanel
  collapsible={true}
  defaultExpand={true}
  titleText="test-title"
>
  <div
    className="panel-collapse"
  >
    <div>
      <div
        className="collapse-header closed tm-collapser"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#eee",
            "height": "40px",
            "padding": "8px 12px",
          }
        }
      >
        <span
          className="collapse-header-arrow-icon fa fa-angle-up"
          style={
            Object {
              "color": "#666",
              "float": "right",
              "fontSize": "20px",
            }
          }
        />
        <span
          className="collapse-header-text"
          style={
            Object {
              "color": "#222",
              "fontSize": "16px",
            }
          }
        >
          test-title
        </span>
      </div>
      <div
        className="collapse"
      >
        <div
          style={
            Object {
              "padding": "8px 0px",
            }
          }
        >
          <div>
            test
          </div>
        </div>
      </div>
    </div>
  </div>
</CollapsiblePanel>
`;

exports[`CollapsiblePanel click on header close by clicking on arrow and open on text 1`] = `
<CollapsiblePanel
  collapsible={true}
  defaultExpand={true}
  titleText="test-title"
>
  <div
    className="panel-collapse"
  >
    <div>
      <div
        className="collapse-header closed tm-collapser"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#eee",
            "height": "40px",
            "padding": "8px 12px",
          }
        }
      >
        <span
          className="collapse-header-arrow-icon fa fa-angle-up"
          style={
            Object {
              "color": "#666",
              "float": "right",
              "fontSize": "20px",
            }
          }
        />
        <span
          className="collapse-header-text"
          style={
            Object {
              "color": "#222",
              "fontSize": "16px",
            }
          }
        >
          test-title
        </span>
      </div>
      <div
        className="collapse"
      >
        <div
          style={
            Object {
              "padding": "8px 0px",
            }
          }
        >
          <div>
            test
          </div>
        </div>
      </div>
    </div>
  </div>
</CollapsiblePanel>
`;

exports[`CollapsiblePanel click on header close by clicking on arrow and open on text 2`] = `
<CollapsiblePanel
  collapsible={true}
  defaultExpand={true}
  titleText="test-title"
>
  <div
    className="panel-collapse"
  >
    <div>
      <div
        className="collapse-header tm-collapser"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#eee",
            "height": "40px",
            "padding": "8px 12px",
          }
        }
      >
        <span
          className="collapse-header-arrow-icon fa fa-angle-down"
          style={
            Object {
              "color": "#666",
              "float": "right",
              "fontSize": "20px",
            }
          }
        />
        <span
          className="collapse-header-text"
          style={
            Object {
              "color": "#222",
              "fontSize": "16px",
            }
          }
        >
          test-title
        </span>
      </div>
      <div
        className="in collapse"
      >
        <div
          style={
            Object {
              "padding": "8px 0px",
            }
          }
        >
          <div>
            test
          </div>
        </div>
      </div>
    </div>
  </div>
</CollapsiblePanel>
`;

exports[`CollapsiblePanel click on header text 1`] = `
<CollapsiblePanel
  collapsible={true}
  defaultExpand={true}
  titleText="test-title"
>
  <div
    className="panel-collapse"
  >
    <div>
      <div
        className="collapse-header closed tm-collapser"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#eee",
            "height": "40px",
            "padding": "8px 12px",
          }
        }
      >
        <span
          className="collapse-header-arrow-icon fa fa-angle-up"
          style={
            Object {
              "color": "#666",
              "float": "right",
              "fontSize": "20px",
            }
          }
        />
        <span
          className="collapse-header-text"
          style={
            Object {
              "color": "#222",
              "fontSize": "16px",
            }
          }
        >
          test-title
        </span>
      </div>
      <div
        className="collapse"
      >
        <div
          style={
            Object {
              "padding": "8px 0px",
            }
          }
        >
          <div>
            test
          </div>
        </div>
      </div>
    </div>
  </div>
</CollapsiblePanel>
`;

exports[`CollapsiblePanel render CollapsiblePanel default 1`] = `
<CollapsiblePanel
  collapsible={true}
  defaultExpand={true}
  titleText="test-title"
>
  <div
    className="panel-collapse"
  >
    <div>
      <div
        className="collapse-header tm-collapser"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#eee",
            "height": "40px",
            "padding": "8px 12px",
          }
        }
      >
        <span
          className="collapse-header-arrow-icon fa fa-angle-down"
          style={
            Object {
              "color": "#666",
              "float": "right",
              "fontSize": "20px",
            }
          }
        />
        <span
          className="collapse-header-text"
          style={
            Object {
              "color": "#222",
              "fontSize": "16px",
            }
          }
        >
          test-title
        </span>
      </div>
      <div
        className="in collapse"
      >
        <div
          style={
            Object {
              "padding": "8px 0px",
            }
          }
        >
          <div>
            test
          </div>
        </div>
      </div>
    </div>
  </div>
</CollapsiblePanel>
`;
